src/tp/tp_segmenter.cpp

```
@@ -94,108 +94,82 @@ TpResult TpSegmenter::create_multi_segments(const Message& message,
+        uint32_t someip_length = 8 + 4 + seg_payload_size;
+        seg_data[4] = static_cast<uint8_t>((someip_length >> 24U) & 0xFFU);
+        seg_data[5] = static_cast<uint8_t>((someip_length >> 16U) & 0xFFU);
+        seg_data[6] = static_cast<uint8_t>((someip_length >> 8U) & 0xFFU);
+        seg_data[7] = static_cast<uint8_t>(someip_length & 0xFFU);
+
+        // Insert 4-byte TP header after SOME/IP header
+        uint32_t offset_units = payload_offset / 16;
+        uint32_t tp_header = (offset_units << 4U) | (more_segments ? 0x01U : 0x00U);
+        seg_data.push_back(static_cast<uint8_t>((tp_header >> 24U) & 0xFFU));
+        seg_data.push_back(static_cast<uint8_t>((tp_header >> 16U) & 0xFFU));
+        seg_data.push_back(static_cast<uint8_t>((tp_header >> 8U) & 0xFFU));
```

Cppcheck (2.21.0)

[style] The function 'get_timestamp' is never used.

(unusedFunction)